tests/test_gene_analysis_integration.py

```diff
@@ -0,0 +1,165 @@
+        alpha=0.1,
+        keep_insignificant=True,
+        minimum_evidence_count=1,
+        minimum_belief=0
+    )
+
+    print(f"Discrete analysis result: {result}")
+    print(f"Discrete analysis result columns: {result.columns if isinstance(result, pd.DataFrame) else 'N/A'}")
+    print(f"Discrete analysis result shape: {result.shape if isinstance(result, pd.DataFrame) else 'N/A'}")
+
+    assert isinstance(result, pd.DataFrame), "Result should be a DataFrame"
+    if result.empty:
+        print("Result DataFrame is empty, skipping further assertions")
+        pytest.skip("Result DataFrame is empty, skipping further assertions")
+    assert "Analysis" in result.columns, "Result should have an 'Analysis' column"
+    assert "p" in result.columns, "Result should have a 'p' column"
+    expected_analyses = {"GO", "WikiPathways", "Reactome", "Phenotype", "INDRA Upstream", "INDRA Downstream"}
+    assert not set(result['Analysis'].unique()).isdisjoint(expected_analyses), \
+        "Result should contain at least one expected analysis type"
+    print("--- Finished test_discrete_analysis_with_real_data ---")
+
+
+def test_signed_analysis_with_real_data(neo4j_client: Neo4jClient):
+    print("\n--- Starting test_signed_analysis_with_real_data ---")
+
+    # Fetch some random genes from the database
+    all_genes = get_random_genes(neo4j_client, 80)  # Assuming get_random_genes is a function you have
+
+    # Split into positive and negative sets
+    positive_genes = {gene_id: gene_name for gene_id, gene_name in list(all_genes.items())[:40]}
+    negative_genes = {gene_id: gene_name for gene_id, gene_name in list(all_genes.items())[40:]}
+
+    print(f"Input positive genes for signed analysis: {positive_genes}")
+    print(f"Input negative genes for signed analysis: {negative_genes}")
+
+    result = signed_analysis(
+        positive_genes,
+        negative_genes,
+        client=neo4j_client,
+        alpha=0.05,
+        keep_insignificant=False,
+        minimum_evidence_count=1,
+        minimum_belief=0
+    )
+
+    # Rest of your test code...
+
+    print(f"Signed analysis result: {result}")
+    print(f"Signed analysis result columns: {result.columns if isinstance(result, pd.DataFrame) else 'N/A'}")
+    print(f"Signed analysis result shape: {result.shape if isinstance(result, pd.DataFrame) else 'N/A'}")
+
+    assert isinstance(result, pd.DataFrame), "Result should be a DataFrame"
+    if result.empty:
+        print("Result DataFrame is empty, skipping further assertions")
+        pytest.skip("Result DataFrame is empty, skipping further assertions")
+    expected_columns = {"curie", "name", "correct", "incorrect", "ambiguous", "binom_pvalue"}
+    assert not expected_columns.isdisjoint(
+        result.columns), f"Result should have at least one of these columns: {expected_columns}"
+    print("--- Finished test_signed_analysis_with_real_data ---")
+
+
+if __name__ == "__main__":
+    pytest.main([__file__])
```

Comment:
I don't believe this is needed, the tests should run anyway if you have pytest installed and run

```sh
pytest tests/test_gene_analysis_integration.py
```

Reply:
Fixed in 720f166c